### PR TITLE
Switch deployment alerts to toast notifications

### PR DESCRIPTION
**Pull request by spadgett**
https://trello.com/c/6WVJdgjL

Updates the following to use toast notifications:

- deploy
- delete deployment
- edit deployment config
- edit resource limits
- edit health checks
- edit yaml (of replication controller)
- deployment rollback
- add/delete/edit env variables
- cancelled deployment
- delete horizontal pod autoscaler

@sg00dwin I've kept your original commit and added another with the changes from my review.

@jwforres PTAL, I've made enough changes to Steve's PR that I think it would be good for you to review.

Replaces https://github.com/openshift/origin-web-console/pull/1594

### CODE REVIEW

File: app/scripts/controllers/deploymentConfig.js [hunk 1]
```diff
@@ -140,21 +141,18 @@ angular.module('openshiftConsole')
...
+                NotificationsService.addNotification({
                   type: "success",
                   // TODO:  improve success alert
                   message: $scope.deploymentConfigName + " was updated."
-                };
+                });
                 $scope.forms.dcEnvVars.$setPristine();
               }, function error(e){
-                $scope.alerts['saveDCEnvVarsError'] = {
+                NotificationsService.addNotification({
```
Comment (jwforres) [inline app/scripts/controllers/deploymentConfig.js]: is there a reason we don't clear this one if you have a successful edit?

File: app/scripts/directives/deleteLink.js [hunk 2]
```diff
@@ -79,12 +79,9 @@ angular.module("openshiftConsole")
             group: 'autoscaling'
           }, hpa.metadata.name, { namespace: scope.projectName })
           .then(function() {
-            showAlert({
-              name: hpa.metadata.name,
-              data: {
+            NotificationsService.addNotification({
                 type: "success",
                 message: "Horizontal Pod Autoscaler " + hpa.metadata.name + " was marked for deletion."
```
Comment (jwforres) [inline app/scripts/directives/deleteLink.js]: not new, but this casing isn't consistent with how we humanize resource types elsewhere. Should we change this one?

File: app/scripts/controllers/deploymentConfig.js (hunk 1 above)
Comment (spadgett) [inline app/scripts/controllers/deploymentConfig.js]: Fixed

File: app/scripts/directives/deleteLink.js (hunk 2 above)
Comment (spadgett) [inline app/scripts/directives/deleteLink.js]: Fixed

**Comment by spadgett** [comment]
@jwforres thanks, updated

**Comment by jwforres** [comment]
[merge]

**Pull request merged**